### ISSUE TITLE
Correct byte array returning

### ISSUE THREAD

**Issue by levBagryansky** [jni-rs/jni-rs#483]
Hi! I have java function returning byte array `byte[]`. I need to call this method from rust, eventually getting "&[u32]". I have not found anywhere in the readme and examples how to correctly call this java method by getting an array of bytes from it in rust. Could you please tell me how to do this, or where it might be described? 
Seems the code below must be there
```
env
            .call_method(
                &self.java_obj,
                "foo",
                "(I)[B",
                &[
                    JValue::Int(my_int as i32),
                ]
            ).unwrap().l().unwrap()
```
but what is next

**Comment by argv-minus-one** [comment]
1. Convert the returned `JObject` to a [`JByteArray`](https://docs.rs/jni/0.21.1/jni/objects/type.JByteArray.html) with `JByteArray::from(the_jobject)`.
2. Get a handle to the bytes using [`JNIEnv::get_array_elements`](https://docs.rs/jni/0.21.1/jni/struct.JNIEnv.html#method.get_array_elements). That handle `Deref`s to `&[i8]`.
3. Use [`cast_slice`](https://docs.rs/bytemuck/1.13.1/bytemuck/fn.cast_slice.html) from the [bytemuck](https://docs.rs/bytemuck/1.13.1/bytemuck/index.html) library to cast `&[i8]` to `&[u32]`.

**Warning:** There is no guarantee that the byte array will be aligned on a four-byte boundary. If it isn't, `bytemuck::cast_slice` will panic. This *might* work, but again, no guarantees. The JVM is allowed to allocate byte arrays without aligning them on a four-byte boundary if it wants to, and there's no way to make sure the array is aligned. You can work around this by casting to `&[[u8; 4]]` instead, and then using `u32::from_be_bytes` on each group of four bytes; this will effectively copy the array (meaning it'll be slow if the array is large), but it will work even if the array is unaligned.

**Warning:** `bytemuck::cast_slice` will also panic if the byte array's length isn't a multiple of four. If your application doesn't already ensure that, then you should use [`bytemuck::try_cast_slice`](https://docs.rs/bytemuck/1.13.1/bytemuck/fn.try_cast_slice.html) instead, and check if it returns [`Err(bytemuck::PodCastError::OutputSliceWouldHaveSlop)`](https://docs.rs/bytemuck/1.13.1/bytemuck/enum.PodCastError.html#variant.OutputSliceWouldHaveSlop). That's the error you'll get if the byte array's length isn't a multiple of four.

Long story short, what you're asking for is surprisingly tricky because of issues with alignment and array size. It can be done, but not as easily as you may have expected.

**Comment by levBagryansky** [comment]
@argv-minus-one Thanks!! `&[u32]` or `&[i8]` is not significant for me. So 
```
let java_array = JByteArray::from(
    java_env
    .call_method(
        &self.java_obj,
        "foo",
        "(I)[B",
        &[
            JValue::Int(v as i32),
        ]
    ).unwrap().l().unwrap());
let size = java_env.get_array_length(&java_array).unwrap();
println!("size = {}", size);
let mut bytes = vec![0; size.try_into().unwrap()];
self.java_env.get_byte_array_region(&java_array, 0, &mut bytes[0..]);
```
works correctly.
